corehq/apps/users/models.py

```diff
@@ -2062,8 +2062,12 @@ def set_groups(self, group_ids):
         touched = []
         faulty_groups = []
         for to_add in desired - current:
-            group = Group.get(to_add)
-            if group.domain != self.domain:
+            try:
+                group = Group.get(to_add)
+            except ResourceNotFound:
+                faulty_groups.append(to_add)
+                continue
+            if group and group.domain != self.domain:
```

Comment:
Why do you need to check `group`? `Group.get` will always either return an object or raise `ResourceNotFound`, right?